### PR TITLE
Acceptance Tests for DocC

### PR DESCRIPTION
**Pull request by Jake-Prickett**
### Short description 📝
A while back I opened a PR adding support for `DocC` documentation archives (https://github.com/tuist/tuist/pull/3982), this PR adds acceptance tests for the changes made to protect the functionality from breaking in the future. Wanted to close the loop and make sure I added this in! 😄 

#### Changes
- Added a documentation build step for leveraging `xcodebuild docbuild` for various scenarios
- Added acceptance test that brings over the `app_with_docc` fixture, generates the project, and verifies the `SlothCreator.doccarchive` exists.

### How to test the changes locally 🧐

Run `./fourier test tuist acceptance projects/tuist/features/docc.feature` in the root of the project.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [noop] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [n/a] In case the PR introduces changes that affect users, the documentation has been updated.
- [n/a] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]
Amazing 👏🚀

**Comment by danyf90** [comment]
Given it's a new feature file it probably needs to be wired somewhere, I don't see it in the list of executed tests

**Comment by Jake-Prickett** [comment]
Looks like we both added it in this morning @danyf90! 😆 Great catch, thank you!

File: projects/tuist/features/docc.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Feature: Tuist docc.
```
Comment (fortmarek) [inline projects/tuist/features/docc.feature:1]:
```suggestion
Feature: Tuist docc
```

File: projects/tuist/features/docc.feature [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+    Scenario: The project is an application with docc documentation (app_with_docc)
+        Given that tuist is available
+        And I have a working directory
+        Then I copy the fixture app_with_docc into the working directory
+        Then tuist generates the project 
+        Then xcodebuild compiles the docc archive for SlothCreator
+        Then a directory Build/Release-iphoneos/SlothCreator.doccarchive exists
```
Comment (fortmarek) [inline projects/tuist/features/docc.feature:9]: This is causing issues on CI - are you sure this is the correct location?

**Review by fortmarek — APPROVED** [review]
Thanks for adding this! We can merge it as soon as the failing CI is fixed ✅

File: projects/tuist/features/docc.feature (hunk 2 above)
Comment (Jake-Prickett) [reply projects/tuist/features/docc.feature:9]: Interesting, it's the correct location locally. I'll look into this one! 👍

**Comment by Jake-Prickett** [comment]
Good to merge @danyf90/@fortmarek 🙌

**Comment by fortmarek** [comment]
Thanks for fixing the CI, LGTM 🚀

**Pull request merged**